packages/miniflare/src/workers/core/entry.worker.ts

```diff
@@ -124,7 +125,9 @@ function getUserRequest(
 
 	request.headers.delete(CoreHeaders.PROXY_SHARED_SECRET);
 	request.headers.delete(CoreHeaders.ORIGINAL_URL);
-	request.headers.delete(CoreHeaders.DISABLE_PRETTY_ERROR);
+	if (env[CoreBindings.STRIP_DISABLE_PRETTY_ERROR]) {
+		request.headers.delete(CoreHeaders.DISABLE_PRETTY_ERROR);
+	}
```

Comment:
We will continue stripping this header by default, except on the ProxyWorker miniflare which will pass the header over to the UserWorker miniflare so that user can disable the pretty error screen when needed.